### ISSUE TITLE
Easy replacing of non-fininte data from magpie objects

### ISSUE THREAD

**Issue by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [pik-piam/magclass#65]
There was an [issue in the remind package](https://github.com/pik-piam/remind/issues/77) related to `NaN` spreading through the data.  A sector share in consumption of an unused energy carrier evaluated to `NaN` (or `Inf`), then emissions calculated based on that share evaluated to `NaN` too, then total sector emissions, than global sector emissions.
I added a function [replace_non_finite()](https://github.com/pik-piam/remind/blob/master/R/replace_non_finite.R) to the `remind` package to replace these with sensible values.  But since it is a general problem and might arise in other packages, too, maybe it should be moved to `magclass` instead?

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
> In addition, the documentation part is misleading as `magclass` supports arithmetic functions out of the box (e.g. the given example of `sum(x,na.rm=TRUE)` should work). If you had a case in which arithmetic functions were not working for magclass objects I would be interested in having a look at it.  

My bad.  I thought it would be possible to replace something like
```R
magpie_4 <- magpie_1 + magpie_2 + magpie_3
```
with
```R
magpie_4 <- sum(magpie_1, magpie_2, magpie_3, na.rm = TRUE)
```
but this did not work, because it (correctly) sums up all elements of the `magpie` objects, instead of adding the elements and preserving the dimensions.
I fixed the description in pik-piam/remind@4054b6f8c6a7298cd73abfb94904c9a5ee76d983

> Could you elaborate a bit on the added value of this function in case I am missing something?

- The same code appears [once and only once](http://wiki.c2.com/?OnceAndOnlyOnce), instead of being scattered all over the code base.
- I think it more likely to get the average `remind` user/developer to remember and to use `replace_non_finite()` then `some_magpie_object[!is.finite(some_magpie_object)] <- replace` when they do a calculation that could result in `NaN`.
- Having this wrapped in a function makes for more readable code (in my opinion), as cause and remedy of the problem are [combined in one statement](https://github.com/pik-piam/remind/blob/master/R/reportEmi.R#L919)
```R
p_share_cco2_DAC <- replace_non_finite(v33_emiDAC / vm_co2capture)
```
instead of being split over two
```R
p_share_cco2_DAC <- v33_emiDAC / vm_co2capture
p_share_cco2_DAC <- p_share_cco2_DAC[!is.finite(p_share_cco2_DAC)] <- 0
```

**Comment by tscheypidi** [comment]
ok, got the point. Can you send your function via PR to the magclass package?
